core/embed/trezorhal/stm32f4/xdisplay/st-7789/display_fb.c

```diff
@@ -165,12 +178,14 @@ display_fb_info_t display_get_frame_buffer(void) {
 
 void display_refresh(void) {
 #ifndef BOARDLOADER
-  wait_for_fb_switch();
-  display_panel_set_window(0, 0, DISPLAY_RESX - 1, DISPLAY_RESY - 1);
 
   if (is_mode_handler()) {
+    display_panel_set_window(0, 0, DISPLAY_RESX - 1, DISPLAY_RESY - 1);
```

Comment:
If a DMA operation is running in the background, there will be two concurrent writers, and the display content will definitely be corrupted. For example, we should stop the DMA before doing this.

Reply:
https://github.com/trezor/trezor-firmware/commit/45db71541d58b4f9dc8e557f914ae4a8d5bc395f